src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -202,6 +207,43 @@ private static AssumeRoleRequest createAssumeRoleRequest(String iamRoleArn) {
                 .withRoleSessionName("Jenkins");
     }
 
+    /**
+     * Provides the {@link AWSSecurityTokenService} for a given {@link AWSCredentials}
+     * @param awsCredentials
+     *
+     * @return {@link AWSSecurityTokenService}
+     */
+    private static AWSSecurityTokenService getAWSSecurityTokenService(AWSCredentials awsCredentials) {
+        ClientConfiguration clientConfiguration = getClientConfiguration();
+        return AWSSecurityTokenServiceClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+                .withClientConfiguration(clientConfiguration)
+                .build();
+    }
+
+    /**
+     * Provides the {@link ClientConfiguration}
+     *
+     * @return {@link ClientConfiguration}
+     */
+    private static ClientConfiguration getClientConfiguration() {
+        Jenkins instance = Jenkins.getInstanceOrNull();
+
+        if (instance == null) {
+            return null;
+        }
+
+        ProxyConfiguration proxy = instance.proxy;
```

Comment:
Instead of the above null check, could use

```suggestion
        ProxyConfiguration proxy = instance != null ? instance.proxy : null;
```